### ISSUE TITLE
tf.nn.avg_pool3d crashes (floating point exception) in NDHWC mode

### ISSUE THREAD

**Issue by DNXie** [tensorflow/tensorflow#42206]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):2.1.0
- Python version:3.6.9
- Bazel version (if compiling from source):N/A
- GCC/Compiler version (if compiling from source):N/A
- CUDA/cuDNN version:N/A
- GPU model and memory:N/A

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
`tf.nn.avg_pool3d` crashes (floating point exception) when `data_format` = NDHWC AND 0 in input.shape, such as `(1,1,0,1,1)`.  Related #42205

**Describe the expected behavior**
expect no crashes
**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.
~~~python
import tensorflow as tf
input=tf.ones((1,1,0,1,1))
tf.nn.avg_pool3d(input=input,ksize=1,strides=1,padding='SAME', data_format='NDHWC')
~~~

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.
~~~python
Floating point exception (core dumped)
~~~

Related #42205

**Comment by amahendrakar** [comment]
@DNXie,
I was able to reproduce the issue with TF v2.1. However, with TF v2.3 I was able to run the code without any issues, please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/c902187922a31c2a1223141677105020/42206.ipynb). Thanks!

**Comment by DNXie** [comment]
@amahendrakar  Seems like it is fixed in the nightly version.

**Comment by amahendrakar** [comment]
@DNXie,
Thank you for the update. Marking this issue as closed since it is resolved. Please feel free to re-open the issue if necessary.
